net/dns/config.go

```diff
@@ -91,6 +91,24 @@ func (c Config) hasDefaultIPResolversOnly() bool {
 	return true
 }
 
+func (c Config) hasHostsWithoutSplitDNSRoutes() bool {
+	for host := range c.Hosts {
```

Comment:
So this is O(N^2) in hosts x routes. Probably okay?

Reply:
commented